johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mappings.java

```diff
@@ -98,8 +119,47 @@ public Boolean isDeduplicateObjects() {
             return deduplicateObjects;
         }
 
+        @Override
+        public MappingType getType() {
+            return MappingType.CLASSMAPPING;
+        }
+
     }
 
+    public static class ParameterizedTypeMapping extends TypeMapping {
+
+        protected ParameterizedTypeMapping(final ParameterizedType type, final AccessMode.Factory factory,
+                               final Map<String, Getter> getters, final Map<String, Setter> setters,
+                               final Adapter<?, ?> adapter,
+                               final ObjectConverter.Reader<?> reader, final ObjectConverter.Writer<?> writer,
+                               final Getter anyGetter, final Method anySetter) {
+            super(type, factory, getters, setters, adapter, reader, writer, anyGetter, anySetter);
+        }
+
+        @Override
+        public MappingType getType() {
+            return MappingType.PARAMETERIZEDTYPEMAPPING;
+        }
+
+    }
+
+    public static class TypeVariableMapping extends TypeMapping {
```

Comment:
same than for parameterized type, we should cache bounds probably as TypeMapping (ClassMapping in the best case), no?

Reply:
again, isn't this what I am already doing ?
In Mappings.class I add both ParameterizedTypeMapping or TypeVariableMapping to the cache, with classes.putIfAbsent